src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1341,5 +1346,51 @@ private Task<bool> InternalReceiveResponseHeadersAsync(WinHttpRequestState state
 
             return state.TcsReceiveResponseHeaders.Task;
         }
+
+        private static Guid LogRequest(HttpRequestMessage request)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Request"))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                _diagnosticSource.Write(
+                    "System.Net.Http.Request",
+                    new
+                    {
+                        Request = request,
```

Comment:
Thanks. Who is "we" here? This doesn't feel right to me. I'd get it at an app layer, but this is a lower level framework component, that will now be publishing private state for anyone to see and mess with. Am I just being paranoid?